### PR TITLE
feat:[CDS-74522]: Initialization steps taking longer than expected and longer than being logged

### PR DESCRIPTION
**Pull request by piyush-bhuwalka**
## Describe your changes

We have added more logs to keep customer in check with whats happening

Tests done:

- Checked if logs are coming or not properly

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50938)

### CODE REVIEW

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger messagecheck

**Comment by piyush-bhuwalka** [comment]
trigger messagecheck

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java [hunk 1]
```diff
@@ -129,6 +132,10 @@ public K8sTaskExecutionResponse executeTaskInternal(
     String cik8BuildTaskParamsStr = cik8InitializeTaskParams.toString();
     ConnectorDetails gitConnectorDetails = cik8InitializeTaskParams.getCik8PodParams().getGitConnector();
 
+    CommandUnitsProgress commandUnitsProgress = CommandUnitsProgress.builder().build();
+    LogCallback executionLogCallback = getExecutionLogCallback(logStreamingTaskClient, commandUnitsProgress);
+    executionLogCallback.saveExecutionLog("Starting job");
```
Comment (anilchowdhury-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:137]: update about podname and namespace

Comment (anilchowdhury-harness) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:137]: Print the delegate logs also to Execution log callback .. wherever is necessary

Comment (piyush-bhuwalka) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:137]: Done

**Comment by piyush-bhuwalka** [comment]
trigger uts

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java [hunk 2]
```diff
@@ -129,11 +132,16 @@ public K8sTaskExecutionResponse executeTaskInternal(
...
 
+    CommandUnitsProgress commandUnitsProgress = CommandUnitsProgress.builder().build();
+    LogCallback executionLogCallback = getExecutionLogCallback(logStreamingTaskClient, commandUnitsProgress);
+
     PodParams podParams = cik8InitializeTaskParams.getCik8PodParams();
     String namespace = podParams.getNamespace();
     String podName = podParams.getName();
     String serviceAccountName = podParams.getServiceAccountName();
 
+    executionLogCallback.saveExecutionLog(format("Starting job to create pod %s on %s namespace", podName, namespace));
```
Comment (harshjain12) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:143]: This is common for all modules. Let us do it only for cd steps

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java [hunk 3]
```diff
@@ -159,11 +167,13 @@ public K8sTaskExecutionResponse executeTaskInternal(
 
         if (cik8InitializeTaskParams.getServicePodParams() != null) {
           for (CIK8ServicePodParams servicePodParams : cik8InitializeTaskParams.getServicePodParams()) {
+            executionLogCallback.saveExecutionLog(format("Creating service for container: %s", servicePodParams));
             log.info("Creating service for container: {}", servicePodParams);
             createServicePod(coreV1Api, namespace, servicePodParams);
           }
         }
 
+        executionLogCallback.saveExecutionLog("Setting up pod spec");
```
Comment (harshjain12) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:176]: why do we need this even for cd ?

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java (hunk 2 above)
Comment (harshjain12) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:143]: why do we need this even for cd ?

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Review by anilchowdhury-harness — APPROVED** [review]

**Review by harshjain12 — APPROVED** [review]

**Comment by piyush-bhuwalka** [comment]
trigger review

**Pull request merged**